### PR TITLE
rust example

### LINKED ISSUE #1: Rust
implement the sample bot in rust

### PR DESCRIPTION
**Pull request by Simon-Laux**
I'm open for suggestions on code improvements.

closes #1

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    let i = interupted.clone();
+
+    ctrlc::set_handler(move || {
+        i.store(true, atomic::Ordering::SeqCst);
+    }).expect("Error setting Ctrl-C handler");
+    
+    ctx.configure();
+
+    // wait for ctrl+c
+    while !interupted.load(atomic::Ordering::SeqCst) {
```
Comment (link2xt) [inline rust/src/main.rs]: "interrupted"

**Comment by link2xt** [comment]
Maybe we should introduce API to set and check "Auto-Submitted" headers on messages. Two echo bots can easily get stuck in infinite loop if they receive an email with a forged "From" field.

**Comment by hpk42** [comment]
On Wed, Feb 19, 2020 at 11:23 -0800, Alexander Krotov wrote:
> Maybe we should introduce API to set and check "Auto-Submitted" headers on messages. Two echo bots can easily get stuck in infinite loop if they receive an email with a forged "From" field.

Yes, a "auto_submitted_header = 0|1" config setting, defaulting to 0.
@link2xt could you create a core issue as i guess you have the RFC handy?

**Comment by link2xt** [comment]
> @link2xt could you create a core issue as i guess you have the RFC handy?

https://github.com/deltachat/deltachat-core-rust/issues/1301

**Pull request merged**